consensus/src/context/difficulty.rs

```diff
@@ -166,30 +169,24 @@ impl DifficultyCache {
         let mut diff_info_popped = Vec::new();
 
         for (new_timestamp, hf) in blocks {
-            self.timestamps.push_back(new_timestamp);
-            self.cumulative_difficulties
-                .push_back(self.cumulative_difficulty() + *difficulties.last().unwrap());
-            if u64::try_from(self.timestamps.len()).unwrap() > self.config.total_block_count() {
+            timestamps.push_back(new_timestamp);
+
+            let last_cum_diff = cumulative_difficulties.back().copied().unwrap_or(1);
+            cumulative_difficulties.push_back(last_cum_diff + *difficulties.last().unwrap());
```

Comment:
I think these are pretty much the same right?

Reply:
Was thinking it would be slower since `.copied()` copies the full `Option<u128>` (size_of = 32 instead of 16) although it seems like the extra ref indirection may be worse:
```rust
use std::hint::black_box as bb;
use std::{collections::VecDeque, time::Instant};

fn main() {
    loop {
        let mut cumulative_difficulties = VecDeque::from([0]);
        let difficulties = vec![0];

        let now = Instant::now();
        for _ in 0..1_000_000_000 {
            // let _ = bb(old(&mut cumulative_difficulties, &difficulties));
            let _ = bb(new(&mut cumulative_difficulties, &difficulties));
        }
        println!("{}", now.elapsed().as_secs_f32());
    }
}

fn old(cumulative_difficulties: &mut VecDeque<u128>, difficulties: &Vec<u128>) {
    let last_cum_diff = bb(cumulative_difficulties.back().copied().unwrap_or(1));
    cumulative_difficulties.push_back(last_cum_diff + *difficulties.last().unwrap());
}

fn new(cumulative_difficulties: &mut VecDeque<u128>, difficulties: &Vec<u128>) {
    let last_cum_diff = bb(cumulative_difficulties.back().unwrap_or(&1));
    cumulative_difficulties.push_back(last_cum_diff + difficulties.last().unwrap());
}
```